bigquery_etl/shredder/bug1751979.py

```diff
@@ -249,3 +249,124 @@
     )
 ) AS payload)
 """
+
+BUG_1751979_MAIN_SUMMARY_V4_UDFS = r"""
+CREATE TEMP FUNCTION sanitize_search_counts_ms(
+  input ARRAY<STRUCT<engine STRING, source STRING, count INT64>>
+) AS (
+  ARRAY(
+    WITH base AS (
+      SELECT
+        -- This reverses the separation logic in main_summary where
+        -- engine and source are parsed from the search_counts key
+        CONCAT(engine, '.', source) AS key,
+        `count`
+      FROM
+        UNNEST(input)
+    ),
+    parsed AS (
+      SELECT
+        *,
+        REGEXP_EXTRACT(key, "([^.]+[.]in-content[:.][^:]+:).*") AS prefix,
+        REGEXP_EXTRACT(key, "[^.]+[.]in-content[:.][^:]+:(.*)") AS code,
+      FROM
+        base
+    ),
+    scrubbed AS (
+      SELECT
+        * REPLACE (
+          IF(
+            prefix IS NULL
+            OR code IN (
+              "none",
+              "other",
+              "hz",
+              "h_",
+              "MOZ2",
+              "MOZ4",
+              "MOZ5",
+              "MOZA",
+              "MOZB",
+              "MOZD",
+              "MOZE",
+              "MOZI",
+              "MOZM",
+              "MOZO",
+              "MOZT",
+              "MOZW",
+              "MOZSL01",
+              "MOZSL02",
+              "MOZSL03",
+              "firefox-a",
+              "firefox-b",
+              "firefox-b-1",
+              "firefox-b-ab",
+              "firefox-b-1-ab",
+              "firefox-b-d",
+              "firefox-b-1-d",
+              "firefox-b-e",
+              "firefox-b-1-e",
+              "firefox-b-m",
+              "firefox-b-1-m",
+              "firefox-b-o",
+              "firefox-b-1-o",
+              "firefox-b-lm",
+              "firefox-b-1-lm",
+              "firefox-b-lg",
+              "firefox-b-huawei-h1611",
+              "firefox-b-is-oem1",
+              "firefox-b-oem1",
+              "firefox-b-oem2",
+              "firefox-b-tinno",
+              "firefox-b-pn-wt",
+              "firefox-b-pn-wt-us",
+              "ubuntu",
+              "ffab",
+              "ffcm",
+              "ffhp",
+              "ffip",
+              "ffit",
+              "ffnt",
+              "ffocus",
+              "ffos",
+              "ffsb",
+              "fpas",
+              "fpsa",
+              "ftas",
+              "ftsa",
+              "newext",
+              "monline_dg",
+              "monline_3_dg",
+              "monline_4_dg",
+              "monline_7_dg"
```

Comment:
The same comments about the source of these codes would be helpful here for clarity/posterity.

Reply:
The intention is to remove this entire file once we have a complete run, so we don't need to worry too much about posterity. The existence of this comment thread I expect to be specific enough to trace history when tracing things through GitHub,